pixeltable/functions/gemini.py

```diff
@@ -242,6 +247,118 @@ def _(model: str) -> str:
     return f'request-rate:veo:{model}'
 
 
+# The vector dimensionality limit imposed by pgvector (note: Pixeltable does not use halfvec yet)
+_MAX_EMBEDDING_DIMENSIONALITY = 2000
+# The highest embedding dimensionality recommended by Google that is within the above limit
+_DEFAULT_EMBEDDING_DIMENSIONALITY = 1536
+
+
+@pxt.udf(resource_pool='request-rate:gemini', batch_size=32)
+async def generate_embedding(
+    input: Batch[str], *, model: str = 'gemini-embedding-001', config: dict | None = None, async_: bool = False
```

Comment:
`config: dict[str, Any]`?

Also, `async_` isn't what Google calls it. I know Google's terminology is confusing but IMO this makes it even more confusing. Maybe call it `use_gemini_batch_api` or some such.